packages/agreement-email-sender/src/services/agreementEmailSenderService.ts

```diff
@@ -1,229 +1,398 @@
+      return sendAgreementActivationEmail(
+        sesEmailManager,
+        readModelService,
+        agreementV2Msg,
+        templateService,
+        agreementEventMailTemplateType.activationSES,
+        logger,
+        sesSenderData,
+        consumer.name,
+        producer.name,
+        recepientsEmails,
+        interopFeBaseUrl
+      );
+    },
+    sendAgreementRejectNotificationEmail: async (
+      agreementV2Msg: AgreementV2,
+      logger: Logger
+    ) => {
+      const agreement = fromAgreementV2(agreementV2Msg);
+      const htmlTemplate = await retrieveHTMLTemplate(
+        agreementEventMailTemplateType.rejection
+      );
 
-  try {
-    const htmlTemplateBuffer = await fs.readFile(
-      `${dirname}/..${templatePath}`
-    );
-    return htmlTemplateBuffer.toString();
-  } catch {
-    throw htmlTemplateNotFound(templatePath);
-  }
-}
+      const eservice = await retrieveAgreementEservice(
+        agreement,
+        readModelService
+      );
 
-function getFormattedAgreementStampDate(
-  agreement: Agreement,
-  stamp: keyof Agreement["stamps"]
-): string {
-  const stampDate = agreement.stamps[stamp]?.when;
+      const producer = await retrieveTenant(
+        agreement.producerId,
+        readModelService
+      );
+      const consumer = await retrieveTenant(
+        agreement.consumerId,
+        readModelService
+      );
 
-  if (stampDate === undefined) {
-    throw agreementStampDateNotFound(stamp, agreement.id);
-  }
-  return dateAtRomeZone(new Date(Number(stampDate)));
+      const rejectionDate = getFormattedAgreementStampDate(
+        agreement,
+        "rejection"
+      );
+
+      const consumerEmail = getLatestTenantMailOfKind(
+        consumer.mails,
+        tenantMailKind.ContactEmail
+      );
+
+      if (!consumerEmail) {
+        logger.warn(
+          `Consumer email not found for agreement ${agreement.id}, skipping email`
+        );
+        return;
+      }
+
+      const mail = {
+        from: { name: sesSenderData.label, address: sesSenderData.mail },
+        subject: `Richiesta di fruizione per ${eservice.name} rifiutata`,
+        to: [consumerEmail.address],
+        body: templateService.compileHtml(htmlTemplate, {
+          interopFeUrl: `https://${interopFeBaseUrl}/ui/it/erogazione/richieste/${agreement.id}`,
+          producerName: producer.name,
+          consumerName: consumer.name,
+          eserviceName: eservice.name,
+          rejectionDate,
+        }),
+      };
+
+      try {
+        logger.info(
+          `Sending email for agreement ${agreement.id} rejection (${emailManagerKind.ses})`
+        );
+        await sesEmailManager.send(mail.from, mail.to, mail.subject, mail.body);
+        logger.info(
+          `Email sent for agreement ${agreement.id} rejection (${emailManagerKind.ses})`
+        );
+      } catch (err) {
+        logger.warn(
+          `Error sending email for agreement ${agreement.id}: ${err}`
+        );
```

Comment:
The message listener don't handle at top level the generic errors, it could terminate the process, I think that it ignore errors by design.

Reply:
Discussed offline with @Viktor-K - let's still throw because this was the previous behavior, it's good that we now catch expected errors so that we can test them, but unexpected errors should still be thrown and not just logged, the `runConsumer` will catch them as it was doing before 👌

Reply:
This commit [b497ac8](https://github.com/pagopa/interop-be-monorepo/pull/1263/commits/b497ac80be289b6c986b2e26ff240bbf3d5e3010) add generic error when email manager (SES and PEC) fails to send emails.